pkg/bgp/path.go

```diff
@@ -0,0 +1,45 @@
+package bgp
+
+import (
+	"github.com/cloudnativelabs/kube-router/v2/pkg"
+	gobgpapi "github.com/osrg/gobgp/v3/api"
+	"k8s.io/klog/v2"
+)
+
+func PathChanged(path *gobgpapi.Path, pl PeerLister, rs pkg.RouteSyncer, tc pkg.TunnelCleaner) error {
```

Comment:
@twz123 - This feels like the right pattern, because it decouples the `routes` package from the `bgp` package by making it implementation agnostic. So I think that making all of the arguments to this function interfaces is the right thing to do, but wanted to see if you had any feedback.